jina/peapods/pea.py

```diff
@@ -360,35 +339,49 @@ def load_plugins(self):
             from ..importer import PathImporter
             PathImporter.add_modules(*self.args.py_modules)
 
-    def loop_teardown(self):
-        """Stop the request loop """
+    def close_zmqlet(self):
+        """Close the zmqlet if exists"""
         if hasattr(self, 'zmqlet'):
             self.zmqlet.close()
 
-    def run(self):
-        """Start the request loop of this BasePea. It will listen to the network protobuf message via ZeroMQ. """
+    def _initialize_executor(self):
         try:
-            # Every logger created in this process will be identified by the `Pod Id` and use the same name
-            self.post_init()
-            self.loop_body()
+            os.environ['JINA_POD_NAME'] = self.name
+            os.environ['JINA_LOG_ID'] = self.args.log_id
+            self.load_plugins()
+            self.load_executor()
+            return self.executor
         except ExecutorFailToLoad:
             self.logger.critical(f'can not start a executor from {self.args.uses}', exc_info=True)
-        except KeyboardInterrupt:
-            self.logger.info('Loop interrupted by user')
-        except SystemError as ex:
-            self.logger.error(f'SystemError interrupted pea loop {repr(ex)}')
-        except DriverError as ex:
-            self.logger.critical(f'driver error: {repr(ex)}', exc_info=True)
-        except zmq.error.ZMQError:
-            self.logger.critical('zmqlet can not be initiated')
-        except Exception as ex:
-            # this captures the general exception from the following places:
-            # - self.zmqlet.recv_message
-            # - self.zmqlet.send_message
-            self.logger.critical(f'unknown exception: {repr(ex)}', exc_info=True)
+
+    def run(self):
+        """Start the request loop of this BasePea. It will listen to the network protobuf message via ZeroMQ. """
+        try:
+            # eventually we could have `executor dump logic` inside executor itself
+            with self._initialize_executor():
```

Comment:
is this line entering the context of executor?.

Reply:
yes, that is the idea